### PR TITLE
Profile Picture V3: Added asset identifiers

### PR DESCRIPTION
**Pull request by vytis**

### CODE REVIEW

File: Source/Model/User/ZMUser.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+let previewProfileAssetIdentifierKey = #keyPath(ZMUser.previewProfileAssetIdentifier)
+let completeProfileAssetIdentifierKey = #keyPath(ZMUser.completeProfileAssetIdentifier)
+
+extension ZMUser {
+    public var previewProfileAssetIdentifier: String {
+        set {
+            modifyValue(newValue, forKey: previewProfileAssetIdentifierKey)
+        }
+        get {
+            return getValue(forKey: previewProfileAssetIdentifierKey)
```
Comment (daehn) [inline Source/Model/User/ZMUser.swift]: Is it necessary to overwrite the getter here? The `getValue` implementation looks like it does the same as the default implementation.

Comment (daehn) [inline Source/Model/User/ZMUser.swift]: But maybe it would need to marked as `@NSManaged` then.

Comment (vytis) [inline Source/Model/User/ZMUser.swift]: It's Swift requirement actually - you can't have custom setter without getter. It seems `@NSManaged` only works without custom accessors, spits out an error in here.

Comment (daehn) [inline Source/Model/User/ZMUser.swift]: Ahh, too bad!

File: Source/Model/User/ZMUser.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import Foundation
+
+let previewProfileAssetIdentifierKey = #keyPath(ZMUser.previewProfileAssetIdentifier)
+let completeProfileAssetIdentifierKey = #keyPath(ZMUser.completeProfileAssetIdentifier)
+
+extension ZMUser {
+    public var previewProfileAssetIdentifier: String? {
+        set {
+            modifyValue(newValue, forKey: previewProfileAssetIdentifierKey)
```
Comment (daehn) [inline Source/Model/User/ZMUser.swift]: Shouldn't the modified key only be set in case the user is the `selfUser`?

**Comment by mikeger** [comment]
@vytis would you add the migration test later?

**Comment by daehn** [comment]
@vytis see https://github.com/wireapp/wire-ios-data-model/wiki/How-to-write-a-Core-Data-migration-test

**Comment by vytis** [comment]
@daehn @mikeger can you take a look at it now - added migration test.

**Pull request merged**